osf/models/base.py

```diff
@@ -28,16 +28,30 @@
 def _check_blacklist(guid):
     return BlackListGuid.objects.filter(guid=guid).exists()
 
+def check_manually_assigned_guid(guid_id, length=5):
+    if not guid_id or guid_id is str or len(guid_id) != length:
```

Comment:
`guid_id is str` will always be false, as this check validates if guid_id is pointer to str:
Elaborating, the only scenario for this check to pass is 

```
guid_id = str
guid_id is str # >>>> True
```

```suggestion
    if not guid_id or isinstance(guid_id, str)  or len(guid_id) != length:
```

Reply:
I had this one `isinstance(guid_id, str)` in my code but `pre-commit` check asked me to change it to `is str` ... and I didn't test it afterwards. Let me take another look.

Reply:
hmm ... this time, it didn't complain ...